modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -262,13 +262,13 @@ public Object visit(PropertyIsLessThanOrEqualTo filter, Object extraData) {
     @Override
     public Object visit(PropertyIsLike filter, Object extraData) {
         BasicDBObject output = asDBObject(extraData);
-        String expr = convert(filter.accept(this, null), String.class);
+        String expr = convert(filter.getExpression().accept(this, null), String.class);
```

Comment:
I expect this makes sense ... since you are handling the remaining parts of the filter below.

Hold on - do you want to check if expression is a Literal first? If it is a literal you have succeeded in encoding it as a mongo query.

If it is not a literal (say a function that generates the pattern used to match against) then your encoding will not work ...